src/ui/handler/scroll_zoom.js

```diff
@@ -356,6 +377,32 @@ class ScrollZoomHandler {
     reset() {
         this._active = false;
     }
+
+    _addScrollZoomBlocker() {
+        if (this._map && !this._container) {
+            this._container = DOM.create('div', 'mapboxgl-scroll-zoom-blocker', this._map._container);
+            let alertMessage = '';
+
+            if (/(Mac|iPad)/i.test(window.navigator.userAgent)) {
+                alertMessage = this._map._getUIString('ScrollZoomBlocker.CmdMessage');
+            } else {
+                alertMessage = this._map._getUIString('ScrollZoomBlocker.CtrlMessage');
+            }
+
+            const textNode = window.document.createTextNode(alertMessage);
+            this._container.appendChild(textNode);
+        }
+    }
+
+    _showBlockerAlert() {
+        if (this._container.style.visibility === 'hidden') this._container.style.visibility = 'visible';
+        this._container.classList.add('mapboxgl-scroll-zoom-blocker-show');
+
+        this._container.addEventListener('transitionend', () => {
+            this._container.classList.remove('mapboxgl-scroll-zoom-blocker-show');
+        }, {once: true});
```

Comment:
OMG I didn't know about `once: true` ❤️

Reply:
I know! it was an exciting discovery. Sadly after much debugging and testing on different browsers, I reverted back to the setTimeout as the transitionend event listener only worked in chrome. This is due to the fact that with how the wheel event is set up, the transition listener gets called multiple times very quickly so in safari and firefox if it doesn't see a change in between transitions, and the transitionend event listener stops firing. If I don't include something like the transitionend event listener or setTimeout to let the dom adjust to the css changes, the alert doesn't show. Is it a deal breaker to use setTimeout?